File: 2021/3/binary_system_test.go
package main

import (
	"testing"
)

func TestGetPowerConsumption(t *testing.T) {

	tests := map[string]struct {
		input []string
		value int
	}{
		"testinput": {
			input: importBinariesFromFile("testinput.txt"),
			value: 198,
		},
	}

	for name, testcase := range tests {
		t.Run(name, func(t *testing.T) {
			eps, gam := getEpsilonAndGammaRates(testcase.input)
			if eps*gam != testcase.value {
				t.Errorf("Incorrect Value; got %d, wanted %d", eps*gam, testcase.value)
			}
		})
	}
}

func TestGetLifeSupportRating(t *testing.T) {

	tests := map[string]struct {
		input []string
		value int
	}{
		"testinput": {
			input: importBinariesFromFile("testinput.txt"),
			value: 230,
		},
	}

	for name, testcase := range tests {
		t.Run(name, func(t *testing.T) {
			ox, co := getOxygenAndCO2Rating(testcase.input)
			if ox*co != testcase.value {
				t.Errorf("Incorrect Value; got %d, wanted %d", ox*co, testcase.value)
			}
		})
	}
}

File: 2021/3/binary_system_test.py
import unittest
import binary_system

class TestGetBinarySystemResults(unittest.TestCase):
    testInput = [line.rstrip() for line in open("2021/2/testinput.txt")]
    def test_get_power_consumption(self):
        val = binary_system.getPowerConsumption(testInput)
        self.assertEqual(val, 198, "Wrong Value")
    
    def test_get_life_support_rating(self):
        val = binary_system.getLifeSupportRating(binaries)
        self.assertEqual(val, 230, "Wrong Value")